### PR TITLE
Submit job ConfigMap with isInteractive flag

### PR DESCRIPTION
**Pull request by omer-dayan**

### CODE REVIEW

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]

File: cmd/job/delete/delete.go [hunk 1]
```diff
@@ -69,46 +70,45 @@ func NewDeleteCommand() *cobra.Command {
...
-			return err
-		}
-
-		if len(trainingTypes) == 0 {
-			runaiTrainer := trainer.NewRunaiTrainer(*kubeClient)
-			job, err := runaiTrainer.GetTrainingJob(jobName, namespaceInfo.Namespace)
-			if err == nil && !job.CreatedByCLI() {
-				return fmt.Errorf("the job '%s' exists but was not created using the runai cli", jobName)
+	for _, trainingType := range trainer.KnownTrainingTypes {
+		maybeConfigMapName := fmt.Sprintf("%s-%s", name, trainingType)
```
Comment (yodarshafrir1) [inline cmd/job/delete/delete.go:80]: what do you think about `configMapPrefix`

File: cmd/job/delete/delete.go [hunk 2]
```diff
@@ -69,46 +70,45 @@ func NewDeleteCommand() *cobra.Command {
...
 
-			return cmdUtil.GetJobDoesNotExistsInNamespaceError(jobName, namespaceInfo)
-		} else if len(trainingTypes) > 1 {
-			return fmt.Errorf("There are more than 1 training jobs with the same name %s, please double check with `%s list | grep %s`. And use `%s delete %s --type` to delete the exact one.",
-				jobName,
-				config.CLIName,
-				jobName,
-				config.CLIName,
-				jobName)
+func deleteTrainingJob(kubeClient *client.Client, jobName string, namespaceInfo types.NamespaceInfo, trainingType string) error {
```
Comment (yodarshafrir1) [inline cmd/job/delete/delete.go:90]: trainingType is unused

File: cmd/job/delete/delete.go [hunk 3]
```diff
@@ -69,46 +70,45 @@ func NewDeleteCommand() *cobra.Command {
...
+	optionalConfigMaps, err := getJobOptionalConfigMaps(jobName, namespaceInfo.Namespace, kubeClient.GetClientset())
+	if err != nil {
+		return err
+	}
+	if len(optionalConfigMaps) == 0 {
```
Comment (yodarshafrir1) [inline cmd/job/delete/delete.go:95]: We will remove that part on the next stages right?

File: pkg/workflow/workflow.go [hunk 4]
```diff
@@ -19,8 +19,8 @@ import (
 *	delete training job with the job name
 **/
 
-func DeleteJob(name, namespace, trainingType string, clientset kubernetes.Interface) error {
-	jobName := GetJobName(name, trainingType)
+func DeleteJob(namespace, configMapName string, clientset kubernetes.Interface) error {
+	jobName := configMapName
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:23]: No reason to use `jobName` here, the name is just misleading

File: pkg/workflow/workflow.go [hunk 5]
```diff
@@ -71,8 +71,12 @@ func GetDefaultValuesFile(environmentValues string) (string, error) {
 	return valueFile.Name(), nil
 }
 
-func GetJobName(name string, trainingType string) string {
-	return fmt.Sprintf("%s-%s", name, trainingType)
+func GetJobName(name string, trainingType string, isInteractive bool) string {
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:74]: GetConfigMapName no?

File: pkg/workflow/workflow.go [hunk 6]
```diff
@@ -214,7 +218,7 @@ func SubmitJob(name string, trainingType string, namespace string, values interf
 	if err != nil {
 		log.Warnf("Creation of job failed. Cleaning up...")
 
-		jobName := GetJobName(name, trainingType)
+		jobName := GetJobName(name, trainingType, isInteractive)
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:221]: configMapName

File: cmd/job/delete/delete.go (hunk 2 above)
Comment (yodarshafrir1) [inline cmd/job/delete/delete.go:90]: This method already looks much better than before, good job 🥇

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]
Looking good, few small comments

File: cmd/job/delete/delete.go (hunk 3 above)
Comment (omer-dayan) [reply cmd/job/delete/delete.go:95]: Yes. we will not talk in terms of config map at all

File: pkg/workflow/workflow.go [hunk 7]
```diff
@@ -218,7 +216,7 @@ func SubmitJob(name, trainingType, namespace string, isInteractive bool, values
 	if err != nil {
 		log.Warnf("Creation of job failed. Cleaning up...")
 
-		jobName := GetJobName(name, trainingType, isInteractive)
+		jobName := GetConfigMapName(name, trainingType, isInteractive)
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:219]: Please rename to `configMapName`

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]

**Review by yodarshafrir1 — APPROVED** [review]
Alt Text

**Pull request merged**